src/main/io/gps.c

```diff
@@ -688,79 +1079,156 @@ void gpsInitUblox(void)
                         break;
                     case UBLOX_MSG_RATE: //Enable UBLOX Messages
-                        if (gpsData.ubloxUsePVT) {
-                            ubloxSetMessageRate(CLASS_NAV, MSG_PVT, 1); // set PVT MSG rate
-                        } else {
-                            ubloxSetMessageRate(CLASS_NAV, MSG_SOL, 1); // set SOL MSG rate
+                        if (gpsData.unitVersion < UBX_VERSION_M8) {
+                            gpsData.state_position++;
+                            break;
+                        }
+                        if (ubloxHasValSetGet()) {
+                            ubloxSetMessageRateValSet(CFG_MSGOUT_UBX_NAV_PVT_I2C, 0);
+                            ubloxSetMessageRateValSet(CFG_MSGOUT_UBX_NAV_PVT_SPI, 0);
+                            ubloxSetMessageRateValSet(CFG_MSGOUT_UBX_NAV_PVT_UART1, 1);
+                            break;
                         }
+                        ubloxSetMessageRate(CLASS_NAV, MSG_NAV_PVT, 1); // set PVT MSG rate
                         break;
                     case UBLOX_MSG_POSLLH:
                         if (gpsData.ubloxUsePVT) {
                             gpsData.state_position++;
-                        } else {
-                           ubloxSetMessageRate(CLASS_NAV, MSG_POSLLH, 1); // set POSLLH MSG rate
+                            break;
                         }
+                        ubloxSetMessageRate(CLASS_NAV, MSG_NAV_POSLLH, 1); // set POSLLH MSG rate
                         break;
                     case UBLOX_MSG_STATUS:
                         if (gpsData.ubloxUsePVT) {
                             gpsData.state_position++;
                         } else {
-                            ubloxSetMessageRate(CLASS_NAV, MSG_STATUS, 1); // set STATUS MSG rate
+                            ubloxSetMessageRate(CLASS_NAV, MSG_NAV_STATUS, 1); // set STATUS MSG rate
                         }
                         break;
                     case UBLOX_MSG_VELNED:
                         if (gpsData.ubloxUsePVT) {
                             gpsData.state_position++;
                         } else {
-                            ubloxSetMessageRate(CLASS_NAV, MSG_VELNED, 1); // set VELNED MSG rate
+                            ubloxSetMessageRate(CLASS_NAV, MSG_NAV_VELNED, 1); // set VELNED MSG rate
+                        }
+                        break;
+                    case UBLOX_MSG_NAV_SAT:
+                        if (gpsData.unitVersion < UBX_VERSION_M9) {
+                            gpsData.state_position++;
+                            break;
+                        }
+                        if (ubloxHasValSetGet()) {
+                            ubloxSetMessageRateValSet(CFG_MSGOUT_UBX_NAV_SAT_I2C, 0);
+                            ubloxSetMessageRateValSet(CFG_MSGOUT_UBX_NAV_SAT_SPI, 0);
+                            ubloxSetMessageRateValSet(CFG_MSGOUT_UBX_NAV_SAT_UART1, 1);
```

Comment:
We should be sending all the sat info every 5 cycles (like 2 lines below below for the UBX-CFG_MSG method).